### PR TITLE
feat: Provide Spring Boot 3 support 

### PR DESCRIPTION
**Pull request by mbfreder**
*Issue #, if available:*  #487

*Description of changes:*
- Updated to Spring 6.0.7, Spring Boot 3.0.4 and Spring Security 6.0.2
- Added Jakarta namespace compatibility

By submitting this pull request

- [x] I confirm that my contribution is made under the terms of the Apache 2.0 license.
- [x] I confirm that I've made a best effort attempt to update all relevant documentation.

### CODE REVIEW

File: aws-serverless-java-container-core/pom.xml [hunk 1]
```diff
@@ -105,9 +122,14 @@
...
             <groupId>org.springframework.security</groupId>
             <artifactId>spring-security-web</artifactId>
-            <version>5.7.7</version>
+            <version>6.0.2</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>jakarta.activation</groupId>
+            <artifactId>jakarta.activation-api</artifactId>
+            <version>2.1.0</version>
```
Comment (deki) [inline aws-serverless-java-container-core/pom.xml:131]: 2.1.1? Why is it needed to have this with compile scope?

File: aws-serverless-java-container-springboot3/pom.xml [hunk 2]
```diff
@@ -8,18 +8,17 @@
...
     <version>1.10-SNAPSHOT</version>
 
     <properties>
-        <spring.version>5.3.26</spring.version>
-        <springboot.version>2.7.10</springboot.version>
-        <springsecurity.version>5.7.7</springsecurity.version>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <spring.version>6.0.7</spring.version>
+        <springboot.version>3.0.4</springboot.version>
```
Comment (deki) [inline aws-serverless-java-container-springboot3/pom.xml:19]: Should be 3.0.5

File: aws-serverless-springboot3-archetype/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -16,7 +16,7 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.7.10</version>
+        <version>3.0.4</version>
```
Comment (deki) [inline aws-serverless-springboot3-archetype/src/main/resources/archetype-resources/pom.xml:19]: 3.0.5

**Review by deki — CHANGES_REQUESTED** [review]
Thank you for your contribution. I made some comments, mostly related to the dependencies used. 

Please make sure to update the pet-store samples in the `samples` folder as well.

File: aws-serverless-java-container-core/pom.xml [hunk 4]
```diff
@@ -17,9 +17,24 @@
 
     <properties>
         <jaxrs.version>2.1.1</jaxrs.version>
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/pom.xml]: Done, thanks.

File: aws-serverless-java-container-core/pom.xml [hunk 5]
```diff
@@ -17,9 +17,24 @@
...
-        <servlet.version>3.1.0</servlet.version>
+        <servlet.version>5.0.0</servlet.version>
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/pom.xml:20]: Updating to 6.0.0 gave me a bunch of 'class not found" issues. That's why I went with 5.0.0.

File: aws-serverless-java-container-core/pom.xml [hunk 6]
```diff
@@ -30,16 +45,16 @@
 
         <!-- https://mvnrepository.com/artifact/javax.servlet/javax.servlet-api -->
         <dependency>
-            <groupId>javax.servlet</groupId>
-            <artifactId>javax.servlet-api</artifactId>
-            <version>${servlet.version}</version>
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
+            <version>5.0.0</version>
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/pom.xml]: Done.

File: aws-serverless-java-container-core/pom.xml [hunk 7]
```diff
@@ -30,16 +45,16 @@
...
         </dependency>
 
         <!-- https://mvnrepository.com/artifact/javax.ws.rs/javax.ws.rs-api -->
         <dependency>
-            <groupId>javax.ws.rs</groupId>
-            <artifactId>javax.ws.rs-api</artifactId>
-            <version>${jaxrs.version}</version>
+            <groupId>jakarta.ws.rs</groupId>
+            <artifactId>jakarta.ws.rs-api</artifactId>
+            <version>3.0.0-M1</version>
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/pom.xml]: Done, thanks.

File: aws-serverless-java-container-core/pom.xml [hunk 8]
```diff
@@ -56,12 +71,15 @@
...
-            <groupId>commons-fileupload</groupId>
-            <artifactId>commons-fileupload</artifactId>
-            <version>1.5</version>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-fileupload2</artifactId>
+            <version>2.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.angus</groupId>
+            <artifactId>angus-mail</artifactId>
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/pom.xml:81]: After doin the namespace change I got this error: 

```
[ERROR] getMimeType_mimeTypeOfCorrectFile_expectMime  Time elapsed: 0.001 s  <<< ERROR!
java.lang.RuntimeException: Provider for jakarta.activation.spi.MimeTypeRegistryProvider cannot be found
        at com.amazonaws.serverless.proxy.internal.servlet.AwsServletContextTest.getMimeType_mimeTypeOfCorrectFile_expectMime(AwsServletContextTest.java:62)

```
After some research I found [this](https://stackoverflow.com/questions/74900271/provider-for-jakarta-activation-spi-mimetyperegistryprovider-cannot-be-found) on stackoverflow. After I added the dependency, the error disappeared.

File: aws-serverless-java-container-core/pom.xml (hunk 1 above)
Comment (mbfreder) [reply aws-serverless-java-container-core/pom.xml]: It's used in the AwsServletContext class. With a test scope, it gives the following errors:
 ```
package jakarta.activation.spi does not exist
cannot find symbol
[ERROR]   symbol:   class MimetypesFileTypeMap
[ERROR]   location: class com.amazonaws.serverless.proxy.internal.servlet.AwsServletContext

```
Updated to 2.1.1

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java [hunk 9]
```diff
@@ -95,6 +96,36 @@ public String getContextPath() {
...
     }
 
+    @Override
+    public void setResponseCharacterEncoding(String encoding) {
+        // do nothing;
+    }
+
+    @Override
+    public String getResponseCharacterEncoding() {
+        return "";
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java]: Updated

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java [hunk 10]
```diff
@@ -95,6 +96,36 @@ public String getContextPath() {
...
+    }
+
+    @Override
+    public void setRequestCharacterEncoding(String encoding) {
+        // do nothing;
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java]: Updated

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java [hunk 11]
```diff
@@ -95,6 +96,36 @@ public String getContextPath() {
...
+    }
+
+    @Override
+    public String getRequestCharacterEncoding() {
+        return "";
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java]: Updated

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java [hunk 12]
```diff
@@ -95,6 +96,36 @@ public String getContextPath() {
...
+    }
+
+    @Override
+    public void setSessionTimeout(int sessionTimeout) {
+        // do nothing;
+    }
+
+    @Override
+    public int getSessionTimeout() {
+        return 1;
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java]: I don't really know what value would be good here. 15?, 30?, 45? does it really matter? I would go with 15 minutes. What do you think?

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java [hunk 13]
```diff
@@ -345,6 +377,11 @@ public ServletRegistration.Dynamic addServlet(String s, Class<? extends Servlet>
         }
     }
 
+    @Override
+    public ServletRegistration.Dynamic addJspFile(String s, String s1) {
+        return null;
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java]: Done

File: aws-serverless-java-container-spring/pom.xml [hunk 14]
```diff
@@ -59,19 +59,17 @@
             <scope>test</scope>
         </dependency>
 
-        <!-- https://mvnrepository.com/artifact/javax.activation/activation -->
         <dependency>
-            <groupId>javax.activation</groupId>
-            <artifactId>activation</artifactId>
-            <version>1.1.1</version>
-            <scope>test</scope>
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-spring/pom.xml:67]: I missed that. Updated.

File: aws-serverless-java-container-spring/pom.xml [hunk 15]
```diff
@@ -80,19 +78,22 @@
...
             <scope>test</scope>
         </dependency>
 
+        <!-- https://mvnrepository.com/artifact/org.glassfish/jakarta.el -->
         <dependency>
             <groupId>org.glassfish</groupId>
-            <artifactId>javax.el</artifactId>
-            <version>3.0.0</version>
+            <artifactId>jakarta.el</artifactId>
+            <version>5.0.0-M1</version>
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-spring/pom.xml]: Thanks, updated.

File: aws-serverless-java-container-springboot3/pom.xml (hunk 2 above)
Comment (mbfreder) [reply aws-serverless-java-container-springboot3/pom.xml]: Updated.

File: aws-serverless-springboot3-archetype/src/main/resources/archetype-resources/pom.xml (hunk 3 above)
Comment (mbfreder) [reply aws-serverless-springboot3-archetype/src/main/resources/archetype-resources/pom.xml]: Updated

File: samples/spring/pet-store/pom.xml [hunk 16]
```diff
@@ -24,9 +24,8 @@
     </licenses>
 
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
-        <spring.version>5.3.26</spring.version>
+        <java.version>17</java.version>
```
Comment (deki) [inline samples/spring/pet-store/pom.xml:27]: `java.version` is a Spring Boot specific property, isn't it? Afaik it only has an effect if the pom extends `spring-boot-starter-parent`.

**Review by deki — CHANGES_REQUESTED** [review]
Let's move all poms to 2.0-SNAPSHOT along with that change. Please note the samples also contain build.gradle files as an alternative that need to be aligned to the Maven poms.

File: samples/spring/pet-store/pom.xml (hunk 16 above)
Comment (mbfreder) [reply samples/spring/pet-store/pom.xml]: You're right. Thanks for pointing that out.

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpApiV2ProxyHttpServletRequest.java [hunk 17]
```diff
@@ -498,6 +492,21 @@ public AsyncContext getAsyncContext() {
...
     }
 
+    @Override
+    public String getRequestId() {
+        return null;
+    }
+
+    @Override
+    public String getProtocolRequestId() {
+        return null;
```
Comment (deki) [inline aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpApiV2ProxyHttpServletRequest.java:502]: return empty string, implement in `AwsHttpServletRequest`

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpServletResponse.java [hunk 18]
```diff
@@ -259,13 +259,13 @@ public void setStatus(int i) {
...
-        if (!canSetHeader()) return;
-        statusCode = i;
-        statusMessage = s;
-    }
+//    @Override
+//    @Deprecated
+//    public void setStatus(int i, String s) {
+//        if (!canSetHeader()) return;
+//        statusCode = i;
+//        statusMessage = s;
```
Comment (deki) [inline aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpServletResponse.java:267]: looks like `statusMessage` variable can be remove completely

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java [hunk 19]
```diff
@@ -44,8 +46,8 @@
     //-------------------------------------------------------------
     // Constants - Public
     // -------------------------------------------------------------
-    public static final int SERVLET_API_MAJOR_VERSION = 3;
-    public static final int SERVLET_API_MINOR_VERSION = 1;
+    public static final int SERVLET_API_MAJOR_VERSION = 5;
```
Comment (deki) [inline aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java:49]: should be 6 now

File: aws-serverless-java-container-spring/pom.xml (hunk 14 above)
[the remark this answers is not in the document]
Comment (deki) [reply aws-serverless-java-container-spring/pom.xml:67]: Should be 2.1.1

File: aws-serverless-java-container-core/pom.xml [hunk 20]
```diff
@@ -95,19 +112,24 @@
...
         <dependency>
             <groupId>org.springframework.security</groupId>
             <artifactId>spring-security-web</artifactId>
-            <version>5.7.7</version>
+            <version>6.0.2</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>jakarta.activation</groupId>
+            <artifactId>jakarta.activation-api</artifactId>
```
Comment (deki) [inline aws-serverless-java-container-core/pom.xml:130]: is it needed here? seems it also works without it...

**Review by deki — CHANGES_REQUESTED** [review]
Let's remove the `java9-plus` profile along with the old `com.sun.activation:jakarta.activation` dependency.

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpApiV2ProxyHttpServletRequest.java (hunk 17 above)
Comment (mbfreder) [reply aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpApiV2ProxyHttpServletRequest.java]: Updated

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpServletResponse.java (hunk 18 above)
Comment (mbfreder) [reply aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsHttpServletResponse.java]: Done

File: aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java (hunk 19 above)
Comment (mbfreder) [reply aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsServletContext.java]: Thanks. Updated

File: aws-serverless-java-container-springboot3/pom.xml [hunk 21]
```diff
@@ -3,33 +3,31 @@
...
     <properties>
-        <spring.version>5.3.26</spring.version>
-        <springboot.version>2.7.10</springboot.version>
-        <springsecurity.version>5.7.7</springsecurity.version>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <spring.version>6.0.8</spring.version>
+        <springboot.version>3.0.5</springboot.version>
+        <springsecurity.version>6.0.2</springsecurity.version>
+        <java.version>17</java.version>
```
[the remark this answers is not in the document]
Comment (mbfreder) [reply aws-serverless-java-container-springboot3/pom.xml]: Right. Updated.

**Pull request merged**